### PR TITLE
Zipcodedataprocessor

### PR DESCRIPTION
**Pull request by k-sharpe**
https://screenshot.googleplex.com/wBEnedp4x1M.png

### CODE REVIEW

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.HashMap;
+import java.util.logging.Logger;
+
+/**
+* This class is used to read zipcode information into memory.
+*/
+public class ZipCodeDataProcessor {
+  private static Logger log = Logger.getLogger("ZipCodeDataProcessor.class");
+  private static DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  public static HashMap<String, ArrayList<Order>> getZipcodeData() {
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:23]: supernit: add empty line above this

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.logging.Logger;
+
+/**
+* This class is used to read zipcode information into memory.
+*/
+public class ZipCodeDataProcessor {
+  private static Logger log = Logger.getLogger("ZipCodeDataProcessor.class");
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:21]: supernit: `Logger.getLogger(ZipCodeDataProcessor.class.getName());`

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        ArrayList orders = new ArrayList<>();
+        orders.add(fromData);
+        ordersToStringMap.put(fromData.getZipcode(), orders);
+      }
+    }
+    return ordersToStringMap;
+  }
+
+  private static Order getOrderFromEntity(Entity data) {
+    Order orderFromEntity = new Order(
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:72]: this is a dupe of the logic in OrderProcessor. I would make this a public helper method in OrderProcessor instead.

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    if (datastoreIsEmpty) {
+      log.info("The datastore is empty. No orders are able to be returned");
+      return ordersToStringMap;
+    }
+    for (Entity dataOrder : ordersFromDatastore.asIterable()) {
+      Order fromData = getOrderFromEntity(dataOrder);
+      if (ordersToStringMap.containsKey(fromData.getZipcode())) {
+        ordersToStringMap.get(fromData.getZipcode()).add(fromData);
+      }
+      else {
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:62]: move up a line:
`} else {`

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
```
Comment (andreahu12) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:37]: nit: remove unnecessary newline

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        ordersToStringMap.put(fromData.getZipcode(), orders);
+      }
+    }
+    return ordersToStringMap;
+  }
+
+  public static HashMap<String, ArrayList<Order>> getZipcodeData(String zipcodeRequest) {
+    HashMap<String, ArrayList<Order>> ordersToStringMap = new HashMap<>();
+    Filter propertyFilter =
+    new FilterPredicate("zipcode", FilterOperator.EQUAL, zipcodeRequest);
```
Comment (andreahu12) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:49]: indent over 2 spaces to the right `new FilterPredicate(...)`

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.lang.RuntimeException
+;/**
```
Comment (andreahu12) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:16]: semicolon on wrong line

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+      if (ordersToStringMap.containsKey(fromData.getZipcode())) {
+        ordersToStringMap.get(fromData.getZipcode()).add(fromData);
+      } else {
+        ArrayList orders = new ArrayList<>();
+        orders.add(fromData);
+        ordersToStringMap.put(fromData.getZipcode(), orders);
+      }
+    }
+    return ordersToStringMap;
+  }
```
Comment (andreahu12) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:64]: I think this is identical to the logic in `getZipcodeData()`? If so, put it in a private helper method and call it in both.

File: postal-injection/src/main/java/com/google/sps/servlets/OrderProcessor.java [hunk 9]
```diff
@@ -27,20 +28,25 @@
...
-            (String) dataOrder.getProperty("shippingLevel"));
-          orders.add(orderFromDatastore);
+          orders.add(getOrderFromEntity(dataOrder));
         }
       }
     return orders;
   }
+
+   public static Order getOrderFromEntity(Entity data) {
+    Order orderFromEntity = new Order(
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/OrderProcessor.java:38]: supernit: can condense to `return new Order (........)`

File: postal-injection/src/main/java/com/google/sps/servlets/OrderProcessor.java [hunk 10]
```diff
@@ -14,9 +14,10 @@
 * This class is used to read order informatino from a datastore service to create java order objects.
 */
 public class OrderProcessor {
+  private static Logger log = Logger.getLogger("OrderProcessor.class");
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/OrderProcessor.java:17]: supernit: `private static final Logger logger = Logger.getLogger(OrderProcessor.class.getName());`

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.lang.RuntimeException
+;/**
+* This class is used to read zipcode information into memory.
+*/
+public class ZipCodeDataProcessor {
+  private static DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:20]: private static final

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+
+  private static HashMap<String, ZipCodeData> orderCreator(HashMap<String, ArrayList<Order>> ordersToStringMap) {
+    HashMap<String, ZipCodeData> result = new HashMap<>();
+    for (String key : ordersToStringMap.keySet()) {
+      result.put(key, new ZipCodeData(key, ordersToStringMap.get(key)));
+    }
+    return result;
+  }
+  
+  private static HashMap<String, ArrayList<Order>> orderCollector(Query query) {
```
Comment (andreahu12) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:45]: nit: rename for clarity `orderCollector` --> `getOrdersPerZipCode`?

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    HashMap<String, ArrayList<Order>> ordersToStringMap = new HashMap<>();
```
Comment (andreahu12) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:46]: nit: rename for clarity `ordersToStringMap` --> `zipCodeToOrdersMap`

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+
+  public static HashMap<String, ZipCodeData> getZipcodeData(String zipcodeRequest) {
+    Filter propertyFilter =
+      new FilterPredicate("zipcode", FilterOperator.EQUAL, zipcodeRequest);
+    Query query = new Query("Order").setFilter(propertyFilter);
+    HashMap<String, ArrayList<Order>> ordersToStringMap = orderCollector(query);
+    return orderCreator(ordersToStringMap);
+  }
+
+  private static HashMap<String, ZipCodeData> orderCreator(HashMap<String, ArrayList<Order>> ordersToStringMap) {
```
Comment (andreahu12) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:37]: nit: rename for clarity `orderCreator` --> `getZipCodeDataPerZipCode`

Comment (andreahu12) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:37]: nit: rename `ordersToStringMap` --> `zipCodeToOrdersMap`

File: postal-injection/src/main/java/com/google/sps/servlets/OrderProcessor.java [hunk 15]
```diff
@@ -27,20 +27,23 @@
...
+  public static Order getOrderFromEntity(Entity data) {
+    return new Order(
+            (String) data.getProperty("orderId"),
+            (String) data.getProperty("zipcode"),
+            (String) data.getProperty("warehouseLatitude"),
+            (String) data.getProperty("warehouseLongitude"),
+            (String) data.getProperty("customerLatitude"),
+            (String) data.getProperty("customerLongitude"),
+            (String) data.getProperty("shipWeightValue"),
+            (String) data.getProperty("distance"),
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/OrderProcessor.java:45]: looking back these values in Order shouldn't have been strings, they should have been floats or something. something to change in a future PR.

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+
+  public static HashMap<String, ZipCodeData> getZipcodeData(String zipcodeRequest) {
+    Filter propertyFilter =
+      new FilterPredicate("zipcode", FilterOperator.EQUAL, zipcodeRequest);
+    Query query = new Query("Order").setFilter(propertyFilter);
+    HashMap<String, ArrayList<Order>> zipCodeToOrdersMap = getOrdersPerZipCode(query);
+    return getZipDataPerZipCode(zipCodeToOrdersMap);
+  }
+
+  private static HashMap<String, ZipCodeData> getZipDataPerZipCode(HashMap<String, ArrayList<Order>> zipCodeToOrdersMap) {
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeDataProcessor.java:37]: nit: line longer than 100 characters

**Pull request merged**